### PR TITLE
[RHDM-319] OpenShift - default Maven user should not be admin

### PR DESCRIPTION
**Pull request by ruromero**
Addresses templates for https://issues.jboss.org/browse/RHDM-319

Signed-off-by: Ruben Romero Montes <rromerom@redhat.com>

### CODE REVIEW

**Comment by bmozaffa** [comment]
@errantepiphany Please note that rhpam70-businesscentral.yaml & rhpam70-full-mysql.yaml are obsolete and not in the final set of templates. I can do a quick "git rm" of those to avoid future confusion, if you want to merge it in a separate PR.

Other than rhpam70-authoring.yaml, the only other template that has authoring, and therefore using maven credentials off business central, is rhpam70-trial-ephemeral.yaml. Since it's already merge, it would be good to update that, @ruromero

**Comment by spolti** [comment]
Just address the David's requests and rebase it, I approved but those need to be addressed first.

**Comment by bmozaffa** [comment]
@ruromero @errantepiphany Unless I'm misunderstanding the change, it is only applicable to rhpam70-trial-ephemeral.yaml at this time.

**Comment by ruromero** [comment]
rebase done, please review @errantepiphany

File: templates/rhpam70-authoring.yaml [hunk 1]
```diff
@@ -188,6 +188,15 @@ parameters:
   description: Fully qualified URL to a Maven repository. If unspecified, will fall back to Business Central service.
   name: MAVEN_REPO_URL
   required: false
+- displayName: Maven repository username
+  description: Username to access the Maven repository.
+  name: MAVEN_REPO_USERNAME
+  required: true
```
Comment (spolti) [inline templates/rhpam70-authoring.yaml]: Same question I did on the rhdm tempaltes, should it be required as we have default values if empty?

Comment (errantepiphany) [reply templates/rhpam70-authoring.yaml]: Circling back, I agree with Filippe that maven repo username should not be required.

**Comment by bmozaffa** [comment]
@ruromero Sorry for dong this again, please reapply your changes to templates/rhpam70-authoring.yaml and templates/rhpam70-trial-ephemeral.yaml as appropriate.

**Comment by ruromero** [comment]
@errantepiphany done, thanks for the review, something happened with the rebase

**Comment by errantepiphany** [comment]
@ruromero Thanks Ruben. All looks good except there's still one unecessary MAVEN_REPO_PATH left in templates/rhpam70-sit.yaml on lines 957-958: https://github.com/ruromero/rhpam-7-openshift-image/blob/ffa171e5c1950383de6f3b713c642a618bbfa6a4/templates/rhpam70-sit.yaml#L957-L958

**Comment by ruromero** [comment]
@errantepiphany removed

File: templates/rhpam70-authoring.yaml [hunk 2]
```diff
@@ -190,13 +190,14 @@ parameters:
   example: http://nexus.nexus-project.svc.cluster.local:8081/nexus/content/groups/public/
   required: false
 - displayName: Maven repository username
-  description: Username to access the Maven repository, if required.
```
Comment (bmozaffa) [inline templates/rhpam70-authoring.yaml]: @ruromero : Why this change, both in the description and the behavior? This is incorrect. Credentials are in fact very rarely needed for accessing maven dependencies. Please open a jira as a major bug and revert this, making sure the original description is maintained and the field is optional across all templates.

CC @errantepiphany @spolti

Comment (errantepiphany) [reply templates/rhpam70-authoring.yaml]: I will fix this in a follow-up PR.

Comment (errantepiphany) [reply templates/rhpam70-authoring.yaml]: Fixed on https://github.com/jboss-container-images/rhpam-7-openshift-image/pull/93 and https://github.com/jboss-container-images/rhpam-7-openshift-image/pull/94
